src/extension/alterschema/linter/multiple_of_default.h

```diff
@@ -0,0 +1,35 @@
+class MultipleOfDefault final : public SchemaTransformRule {
+public:
+  MultipleOfDefault()
+      : SchemaTransformRule{
+            "multiple_of_default",
+            "Setting `multipleOf` to 1 does not add any further constraint"} {};
+
+  [[nodiscard]] auto
+  condition(const sourcemeta::core::JSON &schema,
+            const sourcemeta::core::JSON &,
+            const sourcemeta::core::Vocabularies &vocabularies,
+            const sourcemeta::core::SchemaFrame &,
+            const sourcemeta::core::SchemaFrame::Location &,
+            const sourcemeta::core::SchemaWalker &,
+            const sourcemeta::core::SchemaResolver &) const
+      -> sourcemeta::core::SchemaTransformRule::Result override {
+    return contains_any(
+               vocabularies,
+               {"https://json-schema.org/draft/2020-12/vocab/validation",
+                "https://json-schema.org/draft/2019-09/vocab/validation",
+                "http://json-schema.org/draft-07/schema#",
+                "http://json-schema.org/draft-06/schema#",
+                "http://json-schema.org/draft-04/schema#"}) &&
+           schema.is_object() && schema.defines("multipleOf") &&
+           schema.at("multipleOf").is_number() &&
```

Comment:
I guess you don't need this line if you are already checking the number possibilities afterwards